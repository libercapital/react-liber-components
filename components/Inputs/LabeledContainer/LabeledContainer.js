'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styles = require('./styles');

var _styles2 = require('../styles');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var LabeledContainer = function LabeledContainer(_ref) {
  var children = _ref.children,
      errorMessage = _ref.errorMessage,
      label = _ref.label,
      props = _objectWithoutProperties(_ref, ['children', 'errorMessage', 'label']);

  return _react2.default.createElement(
    'div',
    props,
    _react2.default.createElement(
      _styles.LabelContainer,
      null,
      _react2.default.createElement(
        'span',
        null,
        label
      ),
      _react2.default.createElement(
        _styles2.InputError,
        { visible: Boolean(errorMessage) },
        errorMessage
      )
    ),
    children
  );
};

LabeledContainer.propTypes = {
  children: _propTypes2.default.oneOfType([_propTypes2.default.element, _propTypes2.default.arrayOf(_propTypes2.default.element)]),
  errorMessage: _propTypes2.default.string,
  label: _propTypes2.default.string.isRequired
};

LabeledContainer.defaultProps = {
  children: null,
  errorMessage: ''
};

exports.default = LabeledContainer;