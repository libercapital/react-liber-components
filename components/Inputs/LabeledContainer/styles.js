'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LabelContainer = undefined;

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _Themes = require('../../Themes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var LabelContainer = exports.LabelContainer = _styledComponents2.default.div.withConfig({
  displayName: 'styles__LabelContainer',
  componentId: 'sc-1ije5ti-0'
})(['font-family:\'Roboto\',sans-serif;font-size:16px;font-weight:normal;display:flex;align-content:flex-start;color:', ';& > span{margin-right:15px;}'], function (props) {
  return (0, _Themes.handleThemeFromObject)(props.theme, 'labeledContainer.color', '#a4a4a4');
});