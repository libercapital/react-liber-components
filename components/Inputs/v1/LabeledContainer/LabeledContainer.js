"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("core-js/modules/es6.symbol");

require("core-js/modules/es6.array.index-of");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es6.object.keys");

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _LabeledContainer = require("./LabeledContainer.styles");

var _Base = require("../Base.styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var LabeledContainer = function LabeledContainer(_ref) {
  var children = _ref.children,
      errorMessage = _ref.errorMessage,
      label = _ref.label,
      props = _objectWithoutProperties(_ref, ["children", "errorMessage", "label"]);

  return _react["default"].createElement("div", props, _react["default"].createElement(_LabeledContainer.LabelContainer, null, _react["default"].createElement("span", null, label), _react["default"].createElement(_Base.InputError, {
    visible: Boolean(errorMessage)
  }, errorMessage)), children);
};

LabeledContainer.propTypes = {
  children: _propTypes["default"].oneOfType([_propTypes["default"].element, _propTypes["default"].arrayOf(_propTypes["default"].element)]),
  errorMessage: _propTypes["default"].string,
  label: _propTypes["default"].string.isRequired
};
LabeledContainer.defaultProps = {
  children: null,
  errorMessage: ''
};
var _default = LabeledContainer;
exports["default"] = _default;