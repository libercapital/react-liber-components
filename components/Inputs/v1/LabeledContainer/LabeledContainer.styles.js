"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LabelContainer = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Themes = require("../../../Themes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var LabelContainer = _styledComponents["default"].div.withConfig({
  displayName: "LabeledContainerstyles__LabelContainer",
  componentId: "xef6m9-0"
})(["font-family:'Roboto',sans-serif;font-size:16px;font-weight:normal;display:flex;align-content:flex-start;color:", ";& > span{margin-right:15px;}"], function (props) {
  return (0, _Themes.handleThemeFromObject)(props.theme, 'labeledContainer.color', '#a4a4a4');
});

exports.LabelContainer = LabelContainer;