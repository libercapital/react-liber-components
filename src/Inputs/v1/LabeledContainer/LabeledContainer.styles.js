import styled from 'styled-components';
import { handleThemeFromObject } from '../../../Themes';

export const LabelContainer = styled.div`
  font-family: 'Roboto', sans-serif;
  font-size: 16px;
  font-weight: normal;
  display: flex;
  align-content: flex-start;
  color: ${props => handleThemeFromObject(props.theme, 'labeledContainer.color', '#a4a4a4')};
  & > span {
    margin-right: 15px;
  }
`;
