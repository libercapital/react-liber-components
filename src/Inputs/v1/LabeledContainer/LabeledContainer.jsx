import React from 'react';
import PropTypes from 'prop-types';
import { LabelContainer } from './LabeledContainer.styles';
import { InputError } from '../Base.styles';

const LabeledContainer = ({ children, errorMessage, label, ...props }) => (
  <div {...props}>
    <LabelContainer>
      <span>{label}</span>
      <InputError visible={Boolean(errorMessage)}>{errorMessage}</InputError>
    </LabelContainer>
    {children}
  </div>
);

LabeledContainer.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.arrayOf(PropTypes.element)]),
  errorMessage: PropTypes.string,
  label: PropTypes.string.isRequired,
};

LabeledContainer.defaultProps = {
  children: null,
  errorMessage: '',
};

export default LabeledContainer;
