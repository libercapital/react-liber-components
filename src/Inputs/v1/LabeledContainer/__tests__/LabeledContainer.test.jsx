import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import LabeledContainer from '../LabeledContainer';

describe('Testing LabeledContainer Component', () => {
  configure({ adapter: new Adapter() });

  it('should not render without label prop', () => {
    const error = sinon.stub(console, 'error');
    shallow(<LabeledContainer />);
    expect(error.calledOnce).toBeTruthy();
    error.restore();
  });

  it('should render correctly with label', () => {
    const wrapper = shallow(<LabeledContainer label="Label" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render error when message is passed', () => {
    const wrapper = shallow(<LabeledContainer label="Label" errorMessage="Error" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render children passed', () => {
    const wrapper = shallow(<LabeledContainer label="Label"><div>test</div></LabeledContainer>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
