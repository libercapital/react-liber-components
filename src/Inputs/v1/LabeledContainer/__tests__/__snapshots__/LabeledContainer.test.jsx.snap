// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LabeledContainer Component should render children passed 1`] = `
<div>
  <LabeledContainerstyles__LabelContainer>
    <span>
      Label
    </span>
    <Basestyles__InputError
      visible={false}
    />
  </LabeledContainerstyles__LabelContainer>
  <div>
    test
  </div>
</div>
`;

exports[`Testing LabeledContainer Component should render correctly with label 1`] = `
<div>
  <LabeledContainerstyles__LabelContainer>
    <span>
      Label
    </span>
    <Basestyles__InputError
      visible={false}
    />
  </LabeledContainerstyles__LabelContainer>
</div>
`;

exports[`Testing LabeledContainer Component should render error when message is passed 1`] = `
<div>
  <LabeledContainerstyles__LabelContainer>
    <span>
      Label
    </span>
    <Basestyles__InputError
      visible={true}
    >
      Error
    </Basestyles__InputError>
  </LabeledContainerstyles__LabelContainer>
</div>
`;
